pkg/subscriptions/subscriptions-client.js

```diff
@@ -0,0 +1,416 @@
+
+        /* convert text output to mustache template variables */
+        ret.push({
+            'productName': segmentInfo['Product Name'],
+            'productId': segmentInfo['Product ID'],
+            'version': segmentInfo['Version'],
+            'arch': segmentInfo['Arch'],
+            'status': status,
+            'starts': segmentInfo['Starts'],
+            'ends': segmentInfo['Ends'],
+        });
+    }
+    return ret;
+};
+
+var gettingDetails = false;
+var getDetailsRequested = false;
+function getSubscriptionDetails() {
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    if (gettingDetails) {
+        getDetailsRequested = true;
+        return;
+    }
+    getDetailsRequested = false;
+    gettingDetails = true;
+    cockpit.spawn(['subscription-manager', 'list'],
+            { directory: '/', superuser: "try", environ: ['LC_ALL=C'] })
+        .done(function(output) {
+            client.subscriptionStatus.products = parseMultipleSubscriptions(output);
+            gettingDetails = false;
+            if (getDetailsRequested)
+                getSubscriptionDetails();
+            needRender();
+        })
+        .fail(function(ex) {
+            client.subscriptionStatus.error = ex;
+            console.warn("Subscriptions [getSubscriptionDetails]: couldn't get details: " + ex);
+            if (getDetailsRequested)
+                getSubscriptionDetails();
+            needRender();
+        });
+}
+
+client.registerSystem = function(subscriptionDetails) {
+    var deferred = $.Deferred();
+
+    var args = ['subscription-manager', 'register'];
+    if (subscriptionDetails.url != 'default')
+        args.push('--serverurl', subscriptionDetails.serverUrl);
+
+    // activation keys can't be used with auto-attach
+    if (subscriptionDetails.activationKeys)
+        args.push('--activationkey', subscriptionDetails.activationKeys);
+    else
+        args.push('--auto-attach');
+
+    if (subscriptionDetails.user || subscriptionDetails.password)
+        args.push('--username', subscriptionDetails.user, '--password', subscriptionDetails.password);
+
+    // proxy is optional
+    if (subscriptionDetails.proxy) {
+        args.push('--proxy', subscriptionDetails.proxyServer,
+                  '--proxyuser', subscriptionDetails.proxyUser,
+                  '--proxypass', subscriptionDetails.proxyPass);
+    }
+
+    // only pass org info if user provided it
+    if (subscriptionDetails.org)
+        args.push('--org', subscriptionDetails.org);
+
+    /* TODO DBus API doesn't deliver what we need, so we call subscription manager
+     * without translations and parse the output
+     */
+    var process = cockpit.spawn(args, {
+        directory: '/',
+        superuser: "require",
+        environ: ['LC_ALL=C'],
+        err: "out"
+    });
+
+    // once we start this, we're in an unknown state
```

Comment:
> Why are we in an unknown state? What about the data from the last refresh. Isn't it "valid" until this one returns?

Just calling `register` often changes the system, and we're not yet sure what will happen. If the following update is slow, the dialog might close and the user can look at product details, only for them to disappear after a few seconds to be replaced with the new state.
I think it's better to clearly mark that status as unknown than to present wrong/outdated information.